perun-core/src/main/java/cz/metacentrum/perun/core/impl/ServicesManagerImpl.java

```diff
@@ -131,8 +131,11 @@ public Destination mapRow(ResultSet rs, int i) throws SQLException {
 			destination.setModifiedAt(rs.getString("destinations_modified_at"));
 			destination.setModifiedBy(rs.getString("destinations_modified_by"));
 			try { // do not mind if the column is not in the results
-				if(rs.getString("f_s_des_propagation_type").equals(Destination.PROPAGATIONTYPE_SERIAL)) {
-					destination.setPropagationType(Destination.PROPAGATIONTYPE_SERIAL);
+				String ptype = rs.getString("f_s_des_propagation_type");
+				if(ptype.equals(Destination.PROPAGATIONTYPE_SERIAL) ||
```

Comment:
In regards of need to display info in GUI, please add this optional reading of propagation type to ``RICH_DESTINATION_MAPPER`` too and update select statements to retrieve also ``f_s_des_propagation_type`` column.

Reply:
Revisited, but please review this, especially in RPC.